util/changelog.py

```diff
@@ -80,114 +90,88 @@
         )
         parse.add_argument(
-            '-e', '--end',
-            help="Ending reference for Changelog",
+            '-r', '--repo',
+            help="<org/repo> to generate logs for",
             type=str, action="store",
-            required=True,
+            default='nanocurrency/nano-node',
         )
         parse.add_argument(
-            '--pat',
-            help="Personal Access Token",
+            '-s', '--start',
+            help="Starting reference for Changelog(oldest)",
             type=str, action="store",
-            required=True,
         )
         options = parse.parse_args()
-        self.repo = options.repo.rstrip("/")
-        self.start = options.start
         self.end = options.end
         self.pat = options.pat
-    def __repr__(self):
-        return "<cliArgs(repo='{0}', start='{1}', end='{2}', pat='{3}')>" \
-            .format(self.repo, self.start, self.end, self.pat)
-    def __str__(self):
-        return "Generating a changelog for {0} starting with {1} " \
-            "and ending with {2}".format(self.repo, self.start, self.end)
+        self.repo = options.repo.rstrip("/")
+        self.start = options.start
+
+
 class generateTree:
     def __init__(self, args):
         github = Github(args.pat)
         self.name = args.repo
-        self.repo = github.get_repo(args.repo)
-        self.start = args.start
-        self.end = args.end
-        try:
-            self.startCommit = self.repo.get_commit(args.start)
-        except BaseException:
-            exit("Error finding commit for " + args.start)
-        try:
-            self.endCommit = self.repo.get_commit(args.end)
-        except BaseException:
-            exit("Error finding commit for " + args.end)
-        commits = self.repo.get_commits(sha=self.endCommit.sha)
+        self.repo = github.get_repo(self.name)
+        self.end = self.repo.get_commit(args.end).sha
+        self.start = self.repo.get_commit(args.start).sha
         self.commits = {}
-        self.other_commits = []  # for commits that do not have an associated pull
+        self.other_commits = []
+        commits = self.repo.get_commits(sha=self.end)
         for commit in commits:
-            if commit.sha == self.startCommit.sha:
+            if commit.sha == self.start:
                 break
-            else:
-                message = commit.commit.message.partition('\n')[0]
-                try:
-                    pr_number = int(
-                        message[message.rfind('#')+1:message.rfind(')')])
+            m = commit.commit.message.partition('\n')[0]
+            try:
+                pr_number = int(m[m.rfind('#')+1:m.rfind(')')])
+                pull = self.repo.get_pull(pr_number)
+            except (ValueError, UnknownObjectException):
+                p = commit.get_pulls()
+                if p.totalCount > 0:
+                    pr_number = p[0].number
                     pull = self.repo.get_pull(pr_number)
```

Comment:
@thsfs I found an edge case where if a commit was added directly to the branch and then later included in an open PR somewhere, that PR would still get added into the changelog when it shouldn't. 

To fix this, ensuring whatever pull is found is not in an open state works:

```
pull = self.repo.get_pull(pr_number)
if pull.state == 'open':
                        print(
                            f"Commit is in tree but only associated with open PR {pr_number}: \"{pull.title}\"")
                        self.other_commits.append((commit.sha, m))
                        continue
                        ```

Reply:
Will open this as an issue to improve the `changelog.py`.